File: src/main/java/com/skillcentric/visibility/one/ChildOne.java
package com.skillcentric.visibility.one;

public class ChildOne extends ParentJava {

    public void demo() {

        ParentJava parent = new ParentJava();

        parent.publicMethod();
        parent.protectedMethod();
        parent.packagePrivateMethod();
    }

    @Override
    public String publicMethod() {
        return super.publicMethod();
    }

    @Override
    protected String protectedMethod() {
        return super.protectedMethod();
    }

    @Override
    String packagePrivateMethod() {
        return super.packagePrivateMethod();
    }
}

File: src/main/kotlin/com/skillcentric/visibility/one/childOne.kt
package com.skillcentric.visibility.one

class Child: Parent() {

    fun demo() {

        val parent = Parent()

        parent.publicMethod()
        parent.internalMethod()

//        parent.protectedMethod();
//        parent.privateMethod();

    }

    override fun publicMethod(){
        super.publicMethod()
        println("Some implementation")
    }

    override fun internalMethod() {
        super.internalMethod()
        println("Some implementation")
    }

    override fun protectedMethod() {
        super.protectedMethod()
        println("Some implementation")
    }
}